client/my-sites/checkout/upsell-nudge/professional-email-upsell/index.tsx

```diff
@@ -72,29 +87,64 @@ const ProfessionalEmailUpsell = ( {
 
 	const optionalMailboxFields = [ 'alternativeEmail', 'name' ];
 
+	const changeIntervalLength = ( newIntervalLength: IntervalLength ) => {
+		setSelectedIntervalLength( newIntervalLength );
+		dispatch(
+			recordTracksEvent( 'calypso_professional_email_upsell_nudge_billing_interval_toggle_click', {
+				domain_name: domainName,
+				new_interval: newIntervalLength,
+			} )
+		);
+	};
+
+	/**
+	 * We calculate the price for a year subscription, given how many months we are going to offer for free.
+	 * It takes into account leap years, and months like february.
+	 *
+	 * Example: If we give 3 months for free in February, for a product that cost 35$ yearly then we return the price
+	 * that we need to bill after those 3 months for the rest of the year.
+	 *
+	 * @param productCost
+	 * @param freeMonths
+	 */
+	const getProratedPrice = ( productCost: number, freeMonths: number ) => {
+		const now = new Date();
+		const freeDays = new Date( new Date().setMonth( now.getMonth() + freeMonths ) ).getTime();
+		const nextYearDate = new Date( new Date().setFullYear( now.getFullYear() + 1 ) );
+		const diff = new Date( nextYearDate.getTime() - freeDays );
+		const diffDays =
+			( nextYearDate.getTime() - now.getTime() ) / 86400000 - diff.getTime() / 86400000;
+		const price = ( 365 - diffDays ) / 365;
+		return price * productCost;
+	};
+
 	const getFormattedPrice = (
 		currencyCode: string,
-		productCost: number,
-		productSlug: string
+		intervalLength: IntervalLength,
+		productCost: number
 	): TranslateResult => {
-		if ( productSlug === TITAN_MAIL_MONTHLY_SLUG ) {
+		if ( intervalLength === IntervalLength.MONTHLY ) {
 			return translate( '{{price/}} /mailbox /month', {
 				components: {
 					price: <span>{ formatCurrency( productCost ?? 0, currencyCode ) }</span>,
 				},
 				comment: '{{price/}} is the formatted price, e.g. $20',
 			} );
 		}
-
+		const proratedPrice = getProratedPrice( productCost, 3 );
 		return translate( '{{price/}} /mailbox /year', {
 			components: {
-				price: <span>{ formatCurrency( productCost ?? 0, currencyCode ) }</span>,
+				price: <span>{ formatCurrency( proratedPrice, currencyCode ) }</span>,
 			},
 			comment: '{{price/}} is the formatted price, e.g. $20',
 		} );
 	};
```

Comment:
Sorry @stephanethomas  this is the function that was initially in the upsell, and you suggested to move it to the parent.